### PR TITLE
JBIDE-24963-10 - add new testclasses for CDI 2.0

### PR DESCRIPTION
**Pull request by zcervink**
**JBIDE-24963-10 - Add new testclasses:**
 - WeldBuiltInContextsTestCDI20
 - WeldExcludeTestCDI20
 - WeldParametersAnnotationTestCDI20
 - WeldScanTestCDI20

Signed-off-by: Zbynek Cervinka <zcervink@redhat.com>

**Jenkins:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/107/
**Jira:** https://issues.jboss.org/browse/JBIDE-24963

### CODE REVIEW

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/BeanValidationProviderCDI20.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	
+	private static final String JSR = "JSR-365"; 
+	private static final String DEL_ANNOTATION_INJECT_FROM_METHOD = "Delete annotation @Inject from method";
+	private static final String DEL_ANNOTATION_OBSERVES_FROM_PARAMETER = "Delete annotation @Observes from parameter";
+	private static final String DEL_ANNOTATION_DISPOSES_FROM_PARAMETER = "Delete annotation @Disposes from parameter";
+	private static final String DEL_ANNOTATION_PRODUCES_FROM_METHOD = "Delete annotation @Produces from method";
+	
+	@Override
+	void init() {
+		problems.add(new ValidationProblem(ProblemType.WARNING, ValidationType.NO_BEAN_ELIGIBLE,
```
Comment (odockal) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/BeanValidationProviderCDI20.java]: Same messages as in CDI11 validation provider.

Comment (zcervink) [reply plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/BeanValidationProviderCDI20.java]: Yes, the messages are the same.

**Comment by zcervink** [comment]
testPR

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/BeanValidationProviderCDI20.java (hunk 1 above)
Comment (odockal) [reply plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/BeanValidationProviderCDI20.java]: We agreed, right? So, what shall we do with a repeatable code?

Comment (odockal) [reply plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/BeanValidationProviderCDI20.java]: Hint: we have three classes with same content where only one variable differs and is derived based on CDI version used (1.0, 1.1. 2.0). How can we define the content only once and still has JSR value based on version used?

Comment (zcervink) [reply plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/BeanValidationProviderCDI20.java]: Fixed.

**Comment by zcervink** [comment]
testPR

**Comment by zcervink** [comment]
testPR

**Comment by odockal** [comment]
@zcervink Jenkins job is green?

**Comment by zcervink** [comment]
> @zcervink Jenkins job is green?

No, they are not. I have made some addition changes to fix NPE. Now Jenkins is running. And I have one addition issue - I can not push to this PR, my branch on my fork has up-to-date commit (0887a173ef35), but this PR has old commit. Does your "approval" to this commit has anything to do with this? This issue I am seeing for the first time and I do not how to fix it.

**Comment by zcervink** [comment]
> > @zcervink Jenkins job is green?
> 
> No, they are not. I have made some addition changes to fix NPE. Now Jenkins is running. And I have one addition issue - I can not push to this PR, my branch on my fork has up-to-date commit (0887a173ef35), but this PR has old commit. Does your "approval" to this commit has anything to do with this? This issue I am seeing for the first time and I do not how to fix it.

**Update on this PR**: commit is now up-to-date, PRcheck for the newest commit is running.

**Comment by zcervink** [comment]
@odockal after some fixes the Jenkins jobs are now green: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/164/

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/AbstractValidationProvider.java [hunk 2]
```diff
@@ -19,8 +19,14 @@
 public abstract class AbstractValidationProvider implements IValidationProvider {
 
 	protected List<ValidationProblem> problems = null;
+	protected String jsr;
```
Comment (odockal) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/AbstractValidationProvider.java]: Why are you using this property in parent abstract class? In previous cases you have put it one class down, right?

Comment (zcervink) [reply plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/AbstractValidationProvider.java]: It has a reason - the init() function is called in the parent abstract class, when the jsr is initialized in the class, that extends the parent class. When the init() is called, the jsr would be uninitialized (results in NPE), if you let the jsr be set in constructor of the subclass.

Comment (odockal) [reply plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/AbstractValidationProvider.java]: We agreed that it did not mind (was not testing JSR string in previous PR, so it passes tests) in merged PR. Continue with your approach and we will revise tests that are not taking in account JSR versions.

**Comment by zcervink** [comment]
@odockal PR rebased, conflicts resolved, Jenkins has also passed: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/175/

**Pull request merged**